tools/scoring/score/dimensions/entity_connection_identification.py

```diff
@@ -30,23 +33,30 @@ class EntityConnectionIdentification(Dimension):
   # rather than `translations` to do its calculations
   category = DimensionCategories.COMPLEX
 
-  def _isolate_connections(self, file: DeserializedFile):
+  def _isolate_connections(self, file: DeserializedFile) -> ConnectionsList:
     """Distill individual connections from each entity
     prior to inclusion in sets for global comparison."""
+    Connection = namedtuple('Connection', ['target', 'connection'])
+
     return [
-        tup for tup in (((cloud_device_id, connection)
-                         for connection in entity.connections)
-                        for cloud_device_id, entity in file.items()
-                        if entity.connections is not None) for tup in tup
+        Connection(entity.code, connection) for entity in file.values()
+        if entity.connections is not None for connection in entity.connections
     ]
 
-  def _condense_connections(self, connections):
+  def _condense_connections(self, connections: ConnectionsList, *,
+                            file: DeserializedFile) -> Set[str]:
     """Condense connections into sets of strings
     for easy comparison using intersection."""
-    return set([
-        f'{target} {connection.ctype} {connection.source}'
-        for target, connection in connections
-    ])
+    condensed = set()
+    for cn in connections:
+      cdid_or_code = lambda code_or_guid: next(
+          entity.cloud_device_id or entity.code for entity in file.values()
+          if code_or_guid in [entity.code, entity.guid])
```

Comment:
can you please avoid lambda, my brain gets heated

Reply:
```
for cn in connections:
  foo(code_or_guid)
   
def foo(code_or_guid):
 for entity in file.values():
  if code_or_guid in [entity.code, entity.guid]
     return entity.cloud_device_id

Reply:
https://github.com/google/digitalbuildings/pull/561/commits/9eec8f21541d609a54f52c3bb3207c4fd4aa5832